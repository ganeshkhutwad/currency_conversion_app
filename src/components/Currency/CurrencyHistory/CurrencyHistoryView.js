import React from 'react';
import Table from 'react-bootstrap/Table';
import PropTypes from 'prop-types';

const CurrencyHistoryView = ({ currencyConversionHistory }) => {
    if (currencyConversionHistory.length) {
        return (
            <Table striped bordered hover data-testid="historyRecords">
                <thead>
                    <tr>
                        <th>#</th>
                        <th>Source Currency</th>
                        <th>Source Amount</th>
                        <th>Output Currency</th>
                        <th>Output Amount</th>
                    </tr>
                </thead>
                <tbody>
                    {currencyConversionHistory.map((record, index) => (
                        <tr key={index}>
                            <td>{index + 1}</td>
                            <td>{record.sourceCurrency}</td>
                            <td>{record.sourceAmount}</td>
                            <td>{record.outputCurrency}</td>
                            <td>{record.outputAmount}</td>
                        </tr>
                    ))}
                </tbody>
            </Table>
        );
    } else {
        return (
            <div data-testid="noRecords" style={{ textAlign: 'center' }}>
                No records yet...
            </div>
        );
    }
};

export default CurrencyHistoryView;

CurrencyHistoryView.propTypes = {
    currencyConversionHistory: PropTypes.arrayOf(
        PropTypes.shape({
            sourceCurrency: PropTypes.string.isRequired,
            sourceAmount: PropTypes.string.isRequired,
            outputCurrency: PropTypes.string.isRequired,
            outputAmount: PropTypes.string.isRequired,
        })
    ).isRequired,
};
