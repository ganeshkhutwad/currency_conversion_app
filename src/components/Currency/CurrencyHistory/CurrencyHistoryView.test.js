import React from 'react';
import { render } from '@testing-library/react';
import CurrencyHistoryView from './CurrencyHistoryView';

test('Snapshot For CurrencyHistoryView Component', () => {
    const currencyConversionHistory = [
        {
            sourceCurrency: 'USD',
            sourceAmount: '200',
            outputCurrency: 'EUR',
            outputAmount: '181.40',
        },
    ];
    const { asFragment } = render(
        <CurrencyHistoryView
            currencyConversionHistory={currencyConversionHistory}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});
