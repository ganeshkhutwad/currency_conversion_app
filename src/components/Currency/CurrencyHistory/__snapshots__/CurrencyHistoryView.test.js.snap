// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot For CurrencyHistoryView Component 1`] = `
<DocumentFragment>
  <table
    class="table table-striped table-bordered table-hover"
    data-testid="historyRecords"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Source Currency
        </th>
        <th>
          Source Amount
        </th>
        <th>
          Output Currency
        </th>
        <th>
          Output Amount
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          USD
        </td>
        <td>
          200
        </td>
        <td>
          EUR
        </td>
        <td>
          181.40
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
